Lib/idlelib/debugger.py

```diff
@@ -130,80 +154,187 @@ def make_gui(self):
+        self.status = ttk.Label(controls, text=' ', font=self.status_normal_font)
+        self.status.grid(column=6, row=0, sticky='nw', padx=[25,0])
+        controls.grid(column=0, row=0, sticky='new', pady=[0,6])
+        controls.grid_columnconfigure(7, weight=1)
+
+        self.current_line_img = self.getimage('debug_current.gif')
+        self.regular_line_img = self.getimage('debug_line.gif')
+        self.stack = ttk.Treeview(left, columns=('statement', ),
+                                  height=5, selectmode='browse')
+        self.stack.column('#0', width=100)
+        self.stack.column('#1', width=150)
+        self.stack.tag_configure('error', foreground='red')
+        self.stack.bind('<<TreeviewSelect>>',
+                        lambda e: self.stack_selection_changed())
+        self.stack.bind('<Double-1>', lambda e: self.stack_doubleclick())
+        if windowingsystem == 'aqua':
+            self.stack.bind('<Button-2>', self.stack_contextmenu)
+            self.stack.bind('<Control-Button-1>', self.stack_contextmenu)
+        else:
+            self.stack.bind('<Button-3>', self.stack_contextmenu)
+
+        scroll = ttk.Scrollbar(left, command=self.stack.yview)
+        self.stack['yscrollcommand'] = scroll.set
+        self.stack.grid(column=0, row=2, sticky='nwes')
+        scroll.grid(column=1, row=2, sticky='ns')
+        left.grid_columnconfigure(0, weight=1)
+        left.grid_rowconfigure(2, weight=1)
+
+        right = ttk.Frame(self.pane, padding=5)
+        self.pane.add(right, weight=1)
+        self.vars = ttk.Treeview(right, columns=('value',), height=5,
+                                                selectmode='none')
+        self.locals = self.vars.insert('', 'end', text='Locals',
+                                       open=True)
+        self.globals = self.vars.insert('', 'end', text='Globals',
+                                        open=False)
+        self.vars.column('#0', width=100)
+        self.vars.column('#1', width=150)
+        self.vars.bind('<Motion>', self.mouse_moved_vars)
+        self.vars.bind('<Leave>', self.leave_vars)
+        scroll2 = ttk.Scrollbar(right, command=self.vars.yview)
+        self.vars['yscrollcommand'] = scroll2.set
+        self.vars.grid(column=0, row=0, sticky='nwes')
+        scroll2.grid(column=1, row=0, sticky='ns')
+        right.grid_columnconfigure(0, weight=1)
+        right.grid_rowconfigure(0, weight=1)
+        left.bind('<Configure>', lambda e: self._adjust_layout())
+        self.clear_stack()
+
+
+    def _adjust_layout(self):
+        # if too narrow, move message below buttons
+        if self.left.winfo_width() < 380:
+            self.status.grid(column=0, row=1, columnspan=8, padx=[5,0])
+        else:
+            self.status.grid(column=6, row=0, columnspan=1, padx=[25,0])
+
+    def enable_buttons(self, buttons=None):
+        for key in self.buttons:
+            if buttons is None or not key in buttons:
```

Comment:
```suggestion
            if buttons is None or key not in buttons:
```

Reply:
yes.  Factoid: Microsoft once applied for a patent on 'not in' operator.